libs/i18n/src/lib/services/root-i18n/root-i18n.service.ts

```diff
@@ -1,20 +1,63 @@
-import { Injectable } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
 import { BehaviorSubject, Observable } from 'rxjs';
 
+import { NgxI18nConfiguration } from '../../i18n.types';
+import { NgxI18nConfigurationToken } from '../../tokens';
+
+//TODO: Iben: Once we have a shared lib we should replace the storage and the browser approaches with their corresponding implementations
 @Injectable({ providedIn: 'root' })
-export class RootI18nService {
+export class NgxI18nRootService {
+	/**
+	 * A subject to hold the current language
+	 */
 	private readonly currentLanguageSubject: BehaviorSubject<string> = new BehaviorSubject<string>(
 		undefined
 	);
 
+	constructor(
+		@Inject(PLATFORM_ID) private readonly platformId: string,
+		@Inject(NgxI18nConfigurationToken)
+		private readonly configuration: NgxI18nConfiguration
+	) {}
+
+	/**
+	 * The current language of the application, as an Observable
+	 */
 	public readonly currentLanguage$: Observable<string> =
 		this.currentLanguageSubject.asObservable();
 
+	/**
+	 * The current language of the application
+	 */
 	public get currentLanguage(): string {
 		return this.currentLanguageSubject.getValue();
 	}
 
-	public setCurrentLanguage(language: string) {
+	/**
+	 * Sets the current language of the application and saves it to the local storage. Returns true if the language was set
+	 *
+	 * @param language - The provided language
+	 */
+	public setCurrentLanguage(language: string): void {
+		// Iben: If a language is set that's not part of the available languages, we return a warn
+		if (!this.configuration.availableLanguages.includes(language)) {
```

Comment:
Is it safe to imply that `configuration` will be defined and have `availableLanguages` as property + is it safe to assume that that property will always be Array?

Reply:
As mentioned in private:

- If the configuration isn't defined it isn't provided and it will return a Injection error
- The configuration is typed and cannot not contain availableLanguages, so unless someone uses any this should also not be an issue